From 815300245c92310657228d9df9be013b885e95c2 Mon Sep 17 00:00:00 2001
From: Ryan M Sullivan <Deffpatch@gmail.com>
Date: Fri, 7 Feb 2020 09:55:16 -0800
Subject: [PATCH] Tenth patch

---
 x.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/x.c b/x.c
index 4a0cfcd..e4406c4 100644
--- a/x.c
+++ b/x.c
@@ -660,6 +660,8 @@ setsel(char *str, Time t)
 	XSetSelectionOwner(xw.dpy, XA_PRIMARY, xw.win, t);
 	if (XGetSelectionOwner(xw.dpy, XA_PRIMARY) != xw.win)
 		selclear();
+
+	clipcopy(NULL);
 }
 
 void
-- 
2.25.0

