From fb4fee888e2ad03141e7715c2521af811b7783ae Mon Sep 17 00:00:00 2001
From: Ryan M Sullivan <Deffpatch@gmail.com>
Date: Fri, 7 Feb 2020 11:41:16 -0800
Subject: [PATCH] Forth patch

---
 config.def.h |   3 +
 1 files changed, 3 insertions(+)

diff --git a/config.def.h b/config.def.h
index aee87c3..dc33c75 100644
--- a/config.def.h
+++ b/config.def.h
@@ -165,6 +165,7 @@ static uint forcemousemod = ShiftMask;
  * Internal mouse shortcuts.
  * Beware that overloading Button1 will disable the selection.
  */
+const unsigned int mousescrollincrement = 1;
 static MouseShortcut mshortcuts[] = {
 	/* mask                 button   function        argument       release */
 	{ ShiftMask,            Button4, kscrollup,      {.i = 1} },
@@ -192,8 +193,8 @@ static Shortcut shortcuts[] = {
 	{ TERMMOD,              XK_Y,           selpaste,       {.i =  0} },
 	{ ShiftMask,            XK_Insert,      selpaste,       {.i =  0} },
 	{ TERMMOD,              XK_Num_Lock,    numlock,        {.i =  0} },
-	{ ShiftMask,            XK_Page_Up,     kscrollup,      {.i = -1} },
-	{ ShiftMask,            XK_Page_Down,   kscrolldown,    {.i = -1} },
+	{ ShiftMask ,           Button4,        kscrollup,      {.i =  mousescrollincrement} },
+	{ ShiftMask,            Button5,        kscrolldown,    {.i =  mousescrollincrement} },
 };
 
-- 
2.25.0
